tools/sphere.c

```diff
@@ -0,0 +1,71 @@
+// Mystical Sphere of Slooping
+// Gon
+// 2017-04
+
+inherit NewThingCode;
+inherit AnsiTellCode;
+
+#include <ansi.h>
+
+#define SLOOPEE "/usr/gon/tools/sloopee_shad"
+#define SLOOPER "/usr/gon/tools/slooper_shad"
+
+// Functions
+void do_sloop();
+int do_touch();
+int do_gaze();
+
+void extra_create() {
+    set_short( "A Glowing Sphere" );
+    set_ids( ({"a glowing sphere", "glowing sphere", "sphere", 
+        "a sphere" }) );
+    set_long( "The sphere is glowing a deep purple that "
+        "seems to pulse rhythmically, almost as if the object "
+        "was taking breath. The surface appears to be smooth "
+        "and without any marks. It makes you slightly uneasy "
+        "as you look upon it, almost as if it was aware of your "
+        "presence..." );
+    set_gettable( 0 );
+}
+
+extra_init() {
+    add_action( "do_sloop", "sloop" );
+    add_action( "do_gaze", "gaze" );
+    add_action( "do_touch", "touch" );
+}
+
+// Apply Slooper and Sloopee shadows, respectively
+void do_sloop( string sloopee ) {
```

Comment:
I wouldn't call this parameter 'sloopee', for a couple reasons. First, we don't actually know that it's the sloopee yet because we haven't validated that it contains a valid sloopee name. What if I type 'sloop 7' ? The other reason is  we don't actually want to do much with the string after we validate it. We actually want to pass that string to FINDP() and get the player object, then most of the remaining operations will be using that object. It would be more convenient for us if that object got the name 'sloopee'.

I'd probably just call this param 'arg'.